warehouse/models/staging/payments/littlepay/stg_littlepay__customer_funding_source.sql

```diff
@@ -29,7 +41,47 @@ stg_littlepay__customer_funding_source AS (
         DENSE_RANK() OVER (
             PARTITION BY customer_id
             ORDER BY littlepay_export_ts DESC) AS calitp_customer_id_rank,
-    FROM source
+        -- generate keys now that input columns have been trimmed & cast and files deduped
+        {{ dbt_utils.generate_surrogate_key(['littlepay_export_ts', '_line_number', 'instance']) }} AS _key,
+        {{ dbt_utils.generate_surrogate_key(['funding_source_id', 'customer_id']) }} AS _payments_key,
+    FROM clean_columns
+    {{ qualify_dedupe_full_duplicate_lp_rows() }}
+),
+
+stg_littlepay__customer_funding_source AS (
+    SELECT
+        funding_source_id,
+        funding_source_vault_id,
+        customer_id,
+        bin,
+        masked_pan,
+        card_scheme,
+        issuer,
+        issuer_country,
+        form_factor,
+        principal_customer_id,
+        _line_number,
+        `instance`,
+        extract_filename,
+        ts,
+        littlepay_export_ts,
+        littlepay_export_date,
+        calitp_funding_source_id_rank,
+        calitp_funding_source_vault_id_rank,
+        calitp_customer_id_rank,
+        _key,
+        _payments_key,
+        _content_hash,
+    FROM add_keys_drop_full_dupes
+    -- Some funding sources have incomplete information when first present in data, like missing
```

Comment:
These comments make sense and I *think* this handling seems appropriate but for this data in particular I am having a 💡 moment looking at this -- my only question is whether there's any chance this needs to be handled more like type 2 slowly-changing data. And same question for `product_data`. 

Looking at the data dictionary I see that things like the card's associated zip code and expiration date don't seem to be included so maybe not, but can you expand (probably offline) on which filled-in fields do change? 

Just realized that this table seems a bit dissimilar from the others in that it's more of a dimension than a fact, and I'm not sure if/how we need to handle accordingly.